libs/core/langchain_core/indexes/base.py

```diff
@@ -0,0 +1,369 @@
+            ids: IDs of the documents to get.
+
+        Returns:
+           A list of the requested Documents.
+        """
+        return list(self.lazy_get_by_ids(ids))
+
+    async def aget_by_ids(self, ids: AsyncIterable[str]) -> List[Document]:
+        docs = []
+        async for doc in await self.alazy_get_by_ids(ids):
+            docs.append(doc)
+        return docs
+
+    def delete(
+        self,
+        *,
+        ids: Optional[Iterable[str]] = None,
+        filters: Union[
+            StructuredQuery, Dict[str, Any], List[Dict[str, Any]], None
+        ] = None,
+        **kwargs: Any,
+    ) -> DeleteResponse:
+        """Default implementation only supports deletion by id.
+
+        Override this method if the integration supports deletion by other parameters.
+
+        Args:
+            ids: IDs of the documents to delete. Must be specified.
+            **kwargs: Other keywords args not supported by default. Will be ignored.
+
+        Returns:
+           A dict ``{"succeeded": [...], "failed": [...]}`` with the IDs of the
+           documents that were successfully deleted and the ones that failed to be
+           deleted.
+
+        Raises:
+            ValueError: if ids are not provided.
+        """
+        if ids is None:
+            raise ValueError("Must provide ids to delete.")
+        if filters:
+            kwargs = {"filters": filters, **kwargs}
+        if kwargs:
+            warnings.warn(
+                "Only deletion by ids is supported for this integration, all other "
+                f"arguments are ignored. Received {kwargs=}"
+            )
+        return self.delete_by_ids(ids)
+
+    async def adelete(
+        self,
+        *,
+        ids: Optional[AsyncIterable[str]] = None,
+        filters: Union[
+            StructuredQuery, Dict[str, Any], List[Dict[str, Any]], None
+        ] = None,
+        **kwargs: Any,
+    ) -> DeleteResponse:
+        """Default implementation only supports deletion by id.
+
+        Override this method if the integration supports deletion by other parameters.
+
+        Args:
+            ids: IDs of the documents to delete. Must be specified.
+            **kwargs: Other keywords args not supported by default. Will be ignored.
+
+        Returns:
+           A dict ``{"succeeded": [...], "failed": [...]}`` with the IDs of the
+           documents that were successfully deleted and the ones that failed to be
+           deleted.
+
+        Raises:
+            ValueError: if ids are not provided.
+        """
+        if ids is None:
+            raise ValueError("Must provide ids to delete.")
+        if filters:
+            kwargs = {"filters": filters, **kwargs}
+        if kwargs:
+            warnings.warn(
+                "Only deletion by ids is supported for this integration, all other "
+                f"arguments are ignored. Received {kwargs=}"
+            )
+        return await self.adelete_by_ids(ids)
+
+    def lazy_get(
```

Comment:
What are semantics of lazy_get() -> is this a scan over all available documents?